### DISCUSSION TITLE
NVidiaDVP

### DISCUSSION THREAD

**Discussion by BorettoLuca** [PlusToolkit/PlusLib#959] [Q&A]
Hi,

I was wondering what NVidiaDVP can be used for. Could you please explain?
https://github.com/PlusToolkit/PlusLib/tree/cedf60d0e1c3d009fff33f2d74c3eddeba3ca3e0/src/PlusDataCollection/NVidiaDVP

Thank you

**Comment by adamrankin** [comment]
It is a device for capturing from GPUDirect for Video devices.

**Reply by BorettoLuca** [reply]
Thank you for your reply.
Do you know where I can find any example of configuration file for using it?

**Reply by adamrankin** [reply]
Note, this device is win32 only at this time.

Not at this time. The parameters that are configurable for the device are 
```xml
EnableGPUCPUCopy=TRUE/FALSE
Framelock=TRUE/FALSE
Blocking=TRUE/FALSE
FSAA=int
Field=TRUE/FALSE
ConversionScaleFactor=float float float
ConversionOffsetVector=float float float
ConversionMatrix=float float float float float float float float float
GammaCorrection=float float float
FlipQueueLength=int
GPU=int
VideoFormat=(see list)
DataFormat=(see list)
SyncSource=bi|tri|auto|sdi
Alpha=TRUE/FALSE
cr=int int int int
cb=int int int int
y=int int int
HorizontalDelay=int
VerticalDelay=int
CaptureGPU=int
Sampling=(see list)
BitsPerComponent=(see list)
```

List of video formats
https://github.com/PlusToolkit/PlusLib/blob/9ec1a71d8f11cd9260ad63a209f42e535db920f1/src/PlusDataCollection/NVidiaDVP/vtkPlusNVidiaDVPVideoSourceWin32.cxx#L289

List of data formats
https://github.com/PlusToolkit/PlusLib/blob/9ec1a71d8f11cd9260ad63a209f42e535db920f1/src/PlusDataCollection/NVidiaDVP/vtkPlusNVidiaDVPVideoSourceWin32.cxx#L493

Sampling
https://github.com/PlusToolkit/PlusLib/blob/9ec1a71d8f11cd9260ad63a209f42e535db920f1/src/PlusDataCollection/NVidiaDVP/vtkPlusNVidiaDVPVideoSourceWin32.cxx#L689

BitsPerComponent
https://github.com/PlusToolkit/PlusLib/blob/9ec1a71d8f11cd9260ad63a209f42e535db920f1/src/PlusDataCollection/NVidiaDVP/vtkPlusNVidiaDVPVideoSourceWin32.cxx#L716
